### PR TITLE
Remove Spree namespace from SolidusStarterFrontend assets

### PR DESCRIPTION
**Pull request by gsmendoza**
## Background

Since the starter frontend is copy-pasted into the host application, we don't need to keep the Spree namespace in the generated files. Instead, we can just copy everything at the root level.

## Types of changes

- ~[ ] Bug fix (non-breaking change which fixes an issue)~
- ~[ ] New feature (non-breaking change which adds functionality)~
- ~[ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)~

## Checklist:

- [x] My code follows the code style of this project.
- ~[ ] My change requires a change to the documentation.~
- ~[ ] I have updated the documentation accordingly.~

### CODE REVIEW

**Review by elia — APPROVED** [review]
Maybe for another PR, but we should decide how to namespace the project, be it `solidus-starter-fronted`, `storefront`, or `frontend`, in any case consistency would be great

**Review by kennyadsl — COMMENTED** [review]
Why not getting rid of the `frontend/` part as well? Is it needed?

**Comment by gsmendoza** [comment]
> Why not getting rid of the `frontend/` part as well? Is it needed?

No strong preference for keeping or removing `frontend/`. Let's see what it looks like without `spree/frontend`. Todo.

File: templates/app/assets/javascripts/solidus_starter_frontend.js
Comment (elia) [inline templates/app/assets/javascripts/solidus_starter_frontend.js:8]: Not sure about this, but does the space stops it from being loaded or is just fo alignment?

Comment (gsmendoza) [reply templates/app/assets/javascripts/solidus_starter_frontend.js:8]: Thanks for the catch @elia ! I'll remove the space.

File: templates/app/assets/javascripts/solidus_starter_frontend.js
Comment (kennyadsl) [inline templates/app/assets/javascripts/solidus_starter_frontend.js:10]: This seems the right place to get rid of this `spree` name as well. WDYT? Also, the description above could be probably changed to remove all the spree references.

File: templates/vendor/assets/javascripts/spree/frontend/all.js
Comment (kennyadsl) [inline templates/vendor/assets/javascripts/spree/frontend/all.js:5]: Can't we move these vendor files into non-spree folders?

**Review by kennyadsl — COMMENTED** [review]
👍

**Comment by elia** [comment]
@gsmendoza I added a reference to #234 to the description as that PR should be unnecessary once this is merged 🙌

File: templates/app/assets/javascripts/solidus_starter_frontend.js
Comment (gsmendoza) [reply templates/app/assets/javascripts/solidus_starter_frontend.js:10]: @kennyadsl Yep, I have just checked `core/app/assets/javascripts/spree.js.erb` in Solidus and I found that the Spree JS helpers there are either replaced or not used in Solidus Starter Frontend. I'll remove this require call.

File: templates/vendor/assets/javascripts/spree/frontend/all.js
Comment (gsmendoza) [reply templates/vendor/assets/javascripts/spree/frontend/all.js:5]: @kennyadsl I don't think we can because extensions generated with SolidusDevSupport inject their asset links to these vendor paths. Please see the `add_javascripts` and `add_stylesheets` methods in https://github.com/solidusio/solidus_dev_support/blob/6270b0ba37cae69b7a6d4fcac928e0b4e55fa212/lib/solidus_dev_support/templates/extension/lib/generators/%25file_name%25/install/install_generator.rb.tt.

Comment (kennyadsl) [reply templates/vendor/assets/javascripts/spree/frontend/all.js:5]: If I got it correctly, that is going to work only with the old frontend. When using the new one, ideally no more JS files will be injected this way. @elia what do you think? Too early to remove the support for this?

Comment (gsmendoza) [reply templates/vendor/assets/javascripts/spree/frontend/all.js:5]: I removed `require spree` and didn't encounter any errors when I tested checkout.

Comment (elia) [reply templates/vendor/assets/javascripts/spree/frontend/all.js:5]: TBH, I'm fine with both, on one side very few JS files from extensions will keep working on a front end HTML that deviates conspicuously from the original solidus_frontend, on the other we could try to support the installation process and then let people to fix their app by themselves.

**Review by elia — APPROVED** [review]
Re-approving, we'll probably remove all legacy JS hooks, but let's do it in small steps.

**Pull request merged**
